ims/common/config.py

```diff
@@ -32,82 +33,50 @@ class __BMIConfig:
     # other instance variables are initialized as to not get AttributeErrors
     def __init__(self, filename):
         self.configfile = filename
+        self.bmi = {}
         self.fs = {}
-        self.uid = None
-        self.db_url = None
-        self.iscsi_update_password = None
-        self.iscsi_ip = None
-        self.haas_url = None
-        self.nameserver_ip = None
-        self.nameserver_port = None
-        self.rpcserver_ip = None
-        self.rpcserver_port = None
-        self.bind_ip = None
-        self.bind_port = None
-        self.logs_url = None
-        self.logs_debug = None
-        self.logs_verbose = None
-        self.pxelinux_loc = None
-        self.ipxe_loc = None
-
+        self.db = {}
+        self.logs = {}
+        self.rpc = {}
+        self.http = {}
+        self.iscsi = {}
+        self.tftp = {}
+        self.netiso = {}
+        self.tests = {}
+
+    # Should Check Mandatory sections
     def parse_config(self):
         config = ConfigParser.SafeConfigParser()
         try:
             if not config.read(self.configfile):
                 raise IOError('cannot load ' + self.configfile)
 
-            self.uid = config.get(constants.BMI_CONFIG_SECTION_NAME,
-                                  constants.UID_KEY)
-
-            self.is_service = config.get(constants.BMI_CONFIG_SECTION_NAME,
-                                         constants.SERVICE_KEY) == 'True'
-
-            self.db_url = config.get(constants.DB_CONFIG_SECTION_NAME,
-                                     constants.DB_URL_KEY)
-
-            self.iscsi_update_password = config.get(
-                constants.ISCSI_CONFIG_SECTION_NAME,
-                constants.ISCSI_PASSWORD_KEY)
-
-            self.iscsi_ip = config.get(constants.ISCSI_CONFIG_SECTION_NAME,
-                                       constants.ISCSI_IP_KEY)
-
-            self.haas_url = config.get(constants.HAAS_CONFIG_SECTION_NAME,
-                                       constants.HAAS_URL_KEY)
+            self.fs = dict(config.items(constants.FS_CONFIG_SECTION_NAME))
+            self.netiso = dict(
+                config.items(constants.NETISO_CONFIG_SECTION_NAME))
+            self.iscsi = dict(config.items(constants.ISCSI_CONFIG_SECTION_NAME))
+            self.tftp = dict(config.items(constants.TFTP_CONFIG_SECTION_NAME))
+            self.bmi = dict(config.items(constants.BMI_CONFIG_SECTION_NAME))
+            self.logs = dict(config.items(constants.LOGS_CONFIG_SECTION_NAME))
+            self.http = dict(config.items(constants.HTTP_CONFIG_SECTION_NAME))
+            self.rpc = dict(config.items(constants.RPC_CONFIG_SECTION_NAME))
+            self.db = dict(config.items(constants.DB_CONFIG_SECTION_NAME))
+
```

Comment:
Do you want me to the validation in this PR ? Initially thought of doing it in Future PR.